### PR TITLE
working to add ListRetain

### PR DESCRIPTION
**Pull request by NoSQLKnowHow**
Assigning to Chris for the moment. This just covers the ListRetain, but not the changes to the ListFetch as they are no complete yet.

### CODE REVIEW

**Comment by cprice404** [comment]
@ben-kugler would you mind updating this PR now that you have completed the other work?  I think you are the only one with end-to-end context at this point.

File: docs/develop/api-reference/list-collections.md [hunk 1]
```diff
@@ -13,12 +13,14 @@ This page details the Momento API methods for the [list collection data types](.
...
-| Name            | Type   | Description                                   |
-| --------------- | ------ | --------------------------------------------- |
-| cacheName       | String | Name of the cache.                            |
-| listName        | String | The name of the list item to be retreived.         |
+| Name       | Type   | Description                                                                 |
+|------------|--------|-----------------------------------------------------------------------------|
+| cacheName  | String | Name of the cache.                                                          |
+| listName   | String | The name of the list item to be retrieved.                                  |
+| startIndex | Number | The starting inclusive element of the list to fetch. Default is 0.          |
+| endIndex   | Number | The ending exclusive element of the list to fetch. Default is end of list.  |
```
Comment (cprice404) [inline docs/develop/api-reference/list-collections.md]: do we have any convention for indicating these are optional?  Or should we add (optional) in the description?

File: docs/develop/api-reference/list-collections.md [hunk 2]
```diff
@@ -226,6 +228,33 @@ See [response objects](./response-objects.md) for specific information.
 
 </details>
 
+### ListRetain
+Retains only slice of the list where the start is inclusive and the end is exclusive, what isn't retained, will be dropped from the list.
```
Comment (cprice404) [inline docs/develop/api-reference/list-collections.md]:
```suggestion

Retains only slice of the list where the start is inclusive and the end is exclusive.  The items outside of this range will be dropped from the list.
```

**Review by cprice404 — COMMENTED** [review]
just a few minor things, otherwise lgtm.  @NoSQLKnowHow I think you can also add an approval on this now that it has other committers.

File: docs/develop/api-reference/list-collections.md (hunk 1 above)
Comment (NoSQLKnowHow) [reply docs/develop/api-reference/list-collections.md]: Which ones are optional?

Comment (cprice404) [reply docs/develop/api-reference/list-collections.md]: startIndex and endIndex

**Review by cprice404 — APPROVED** [review]

**Pull request merged**